featuretools/entityset/entity.py

```diff
@@ -284,40 +281,42 @@ def delete_variables(self, variable_ids):
             self.variables.remove(v)
 
     def set_time_index(self, variable_id, already_sorted=False):
-        # check time type
-        if not isinstance(self.df, pd.DataFrame) or self.df.empty:
-            time_to_check = vtypes.DEFAULT_DTYPE_VALUES[self[variable_id]._default_pandas_dtype]
-        else:
-            time_to_check = self.df[variable_id].iloc[0]
-        time_type = _check_time_type(time_to_check)
-
-        if time_type is None:
-            raise TypeError("%s time index not recognized as numeric or"
-                            " datetime" % (self.id))
+        self.time_index = variable_id
+        self._already_sorted = already_sorted
+        self._check_time_index()
```

Comment:
I think we should leave the `set_time_index` code largely unchanged, aside from moving the `time_type` logic. 

I don't think an `_already_sorted` attribute is necessary, both Entity and Woodwork Table sort by default.

Reply:
✅  df5eb515